sdk/feature-management-applicationinsights-browser/src/telemetry.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { EvaluationResult } from "@microsoft/feature-management";
+import { ApplicationInsights } from "@microsoft/applicationinsights-web";
+import { IEventTelemetry } from "@microsoft/applicationinsights-web";
+
+/**
+ * Creates a telemetry publisher that sends feature evaluation events to Application Insights.
+ * @param client The Application Insights telemetry client.
+ * @returns A callback function that takes an evaluation result and tracks an event with the evaluation details.
+ */
+export function createTelemetryPublisher(client: ApplicationInsights): (event: EvaluationResult) => void {
+    return (event: EvaluationResult) => {
+        const eventProperties = {
+            "FeatureName": event.feature?.id,
+            "Enabled": event.enabled.toString(),
+            // Ensure targetingId is string so that it will be placed in customDimensions
+            "TargetingId": event.targetingId?.toString(),
+            "Variant": event.variant?.name,
+            "VariantAssignmentReason": event.variantAssignmentReason,
+        };
+
+        const metadata = event.feature?.telemetry?.metadata;
+        if (metadata) {
+            for (const key in metadata) {
+                if (!(key in eventProperties)) {
+                    eventProperties[key] = metadata[key];
+                }
+            }
+        }
+
+        client.trackEvent({ name: "FeatureEvaluation" }, eventProperties);
```

Comment:
I'd rather have this call our `trackEvent` function and let it append the `TargetingId` there.

Reply:
I prefer to keep the current implementation. It will be more clear that we are following the [FeatureEvaluationEvent](https://github.com/microsoft/FeatureManagement/tree/main/Schema/FeatureEvaluationEvent) schema.